### DISCUSSION TITLE
Use EntityFramework instead of PetaPoco

### DISCUSSION THREAD

**Discussion by DanielPeinhopf** [umbraco/Umbraco-CMS#10780] [Features and ideas]
Sorry if this question has already been asked, but were there any discussions about switching to Entity Framework instead of PetaPoco for the .NET Core Version of Umbraco?

Entity Framework is "the" ORM for .NET Core and provides support for multiple database systems.
Are there any plans?

**Comment by nul800sebastiaan** [comment]
You haven't missed anything since we haven't discussed it. Right now there are no plans! 🙂

**Reply by DanielPeinhopf** [reply]
Thanks for your reply.
I'll keep my fingers crossed for the future 🤞

The main reason why i'am asking this is "Postgres Support".

**Reply by nul800sebastiaan** [reply]
Yeah sure, we'd love to support more database types, it's not very high on our priorities list at the moment and switching away from NPoco will be large blocker from people wanting to go from v8 to v9. If the whole data layer is going to change, I expect it to be a few years out.

**Reply by DanielPeinhopf** [reply]
Ok thank you for the info.
I've found out that also the Express edition of MS SQL Server is available for linux in the meantime, so there is a free option to run Umbraco on linux.
I can't wait for the final .NET Core version of umbraco but the latest beta looks promising. Keep up the good work guys 👍.
